src/client/connection.ts

```diff
@@ -55,29 +43,37 @@ export const eventEmitter = new EventEmitter();
  */
 export let channelPromise: Promise<ChannelClient>;
 
-if (fin.Window.me.uuid !== SERVICE_IDENTITY.uuid || fin.Window.me.name !== SERVICE_IDENTITY.name) {
-    channelPromise = typeof fin === 'undefined' ?
-        Promise.reject(new Error('fin is not defined. The openfin-notifications module is only intended for use in an OpenFin application.')) :
-        fin.InterApplicationBus.Channel.connect(SERVICE_CHANNEL, {payload: {version: PACKAGE_VERSION}}).then((channel: ChannelClient) => {
-            // Register service listeners
-            channel.register('WARN', (payload: any) => console.warn(payload));
-            channel.register('event', (event: EventPayload<NotificationsEvent>) => {
-                // Rehydrate the date object on the event payload
-                const parsedEvent = {
-                    ...event,
-                    notification: {
-                        ...event.notification,
-                        date: new Date(event.notification.date)
-                    }
-                };
+if (typeof fin !== 'undefined') {
+    getServicePromise();
+}
+
+export function getServicePromise(): Promise<ChannelClient> {
```

Comment:
Copied the `getServicePromise` mechanism over from FDC3 - the un-guarded use of `fin` above was causing issues in the integration tests.